l5kit/l5kit/data/zarr_utils.py

```diff
@@ -1,66 +1,223 @@
-            print(f"input frames size: {len(input_dataset.frames)}")
-            print(f"input agents size: {len(input_dataset.agents)}")
-            print(f"input tl_faces size: {len(input_dataset.tl_faces)}")
-
-        frame_offset = len(output_dataset.frames)
-        new_scenes = np.zeros(len(input_dataset.scenes), dtype=SCENE_DTYPE)
-
-        for i, scene in enumerate(input_dataset.scenes):  # add new scenes to zarr
-            scene["frame_index_interval"] = scene["frame_index_interval"] + frame_offset
-            new_scenes[i] = scene
-        output_dataset.scenes.append(new_scenes)
-
-        agent_offset = len(output_dataset.agents)
-        tl_faces_offset = len(output_dataset.tl_faces)
-
-        new_frames = np.zeros(len(input_dataset.frames), dtype=FRAME_DTYPE)
-        for i, frame in enumerate(input_dataset.frames):  # add new frames to the zarr
-            frame["agent_index_interval"] = frame["agent_index_interval"] + agent_offset
-            frame["traffic_light_faces_index_interval"] = frame["traffic_light_faces_index_interval"] + tl_faces_offset
-            new_frames[i] = frame
-        output_dataset.frames.append(new_frames)
-
-        output_dataset.agents.append(input_dataset.agents)  # add new agents to the zarr
-        output_dataset.tl_faces.append(input_dataset.tl_faces)  # add new traffic light faces to the zarr
-
-    if verbose:
-        print(f"output scenes size: {len(output_dataset.scenes)}")
-        print(f"output frames size: {len(output_dataset.frames)}")
-        print(f"output agents size: {len(output_dataset.agents)}")
-        print(f"output tl_faces size: {len(output_dataset.tl_faces)}")
+        _append_zarr_subset(input_dataset, output_dataset, 0, len(input_dataset.scenes), cur_num_els)
+        cur_num_els += Counter(num_els_valid_zarrs[idx])
+
+
+def zarr_split(input_zarr: str, output_path: str, split_infos: List[dict]) -> List[Tuple[int, int]]:
+    """
+    Split the input zarr into many zarrs. Names and sizes can be passed using the split_infos arg.
+
+    Args:
+        input_zarr (str): path of the original zarr
+        output_path (str): base destination path
+        split_infos (List[dict]): list of dict. Each element should have `name` (final path is output_path+name)
+        and `split_size_GB` with the size of the split. Last element must have `split_size_GB` set to -1 to collect
+        the last part of the input_zarrr.
+
+    Returns:
+        List[Tuple[int, int]]: indices of scenes where a split occurred
+    """
+    input_dataset = ChunkedDataset(input_zarr)
+    input_dataset.open()
+
+    assert len(split_infos) > 0
+    assert split_infos[-1]["split_size_GB"] == -1, "last split element should have split_size_GB equal to -1"
+
+    # compute the size of the input_dataset in GB
+    size_input_gb = _compute_path_size(input_zarr) / GIGABYTE
+    num_scenes_input = len(input_dataset.scenes)
+
+    # ensure we can fit everything
```

Comment:
Doesn't the following assert and thus this comment check something more like 'Did the user define splits which are too large?' (kind of the opposite of 'we can fit everything'

Reply:
It's probably the same thing seen from two different points of view xD 
I'll change for clarity :)